File: api/src/main/java/dev/vibris/api/CaptureResult.java
package dev.vibris.api;

import java.nio.file.Path;
import java.util.Map;

public record CaptureResult(long frameId, Map<String, Path> artifacts) {
    public CaptureResult {
        if (frameId < 0) throw new IllegalArgumentException("frameId must not be negative");
        artifacts = Map.copyOf(artifacts);
    }
}

File: api/src/main/kotlin/dev/vibris/api/CaptureResult.kt
package dev.vibris.api

@JvmRecord
data class CaptureResult(
    val frameId: Long,
    @field:DefensiveSnapshot val groups: List<ArtifactGroup>,
) {
    init {
        require(frameId >= 0) { "frameId must not be negative" }
    }

    @JvmRecord
    data class ArtifactGroup(
        val name: String,
        val resource: ResourceCatalog.ResourceDescriptor,
        @field:DefensiveSnapshot val artifacts: List<CapturedArtifact>,
    )

    @JvmRecord
    data class CapturedArtifact(
        val fileName: String,
        val format: CapturePlan.ArtifactFormat,
        val role: CapturePlan.ArtifactRole,
        val subresourceIndex: Int?,
    )
}
